File: public/downloads/puce.py
from requests import get

class Puce:
    _base = ''
    _url = ''
    _test = []

    def __init__(self, key = ''):
        self._base = 'https://cosmox.ga/api/{}'.format(key)


    def account_create_test(self, name='', email='', password='', pin='', code=''):
        self._test = {'account_create': True}

        return self.account_create(name, email, password, pin, code)


    def account_create(self, name = '', email = '', password = '', pin = '', code = ''):
        self._url = '{}/account/create/{}/{}/{}/{}/{}'.format(self._base, name, email, password, pin, code)

        if 'account_create' in self._test:
            return self._url

        return self._curl()


    def account_change_test(self, name='', email='', password='', pin='', code=''):
        self._test = {'account_change': True}

        return self.account_change(name, email, password, pin, code)


    def account_change(self, name = '', email = '', password = '', pin = '', code = ''):
        self._url = '{}/account/change/{}/{}/{}/{}/{}'.format(self._base, name, email, password, pin, code)

        if 'account_change' in self._test:
            return self._url

        return self._curl()


    def account_change_name_test(self, name = ''):
        self._test = {'account_change_name': True}

        return self.account_change_name(name)


    def account_change_name(self, name = ''):
        self._url = '{}/account/change/name/{}'.format(self._base, name)

        if 'account_change_name' in self._test:
            return self._url

        return self._curl()


    def account_change_email_test(self, email = ''):
        self._test = {'account_change_email': True}

        return self.account_change_email(email)


    def account_change_email(self, email=''):
        self._url = '{}/account/change/email/{}'.format(self._base, email)

        if 'account_change_email' in self._test:
            return self._url

        return self._curl()


    def account_change_password_test(self, passowrd = ''):
        self._test = {'account_change_passowrd': True}

        return self.account_change_password(passowrd)


    def account_change_password(self, password = ''):
        self._url = '{}/account/change/password/{}'.format(self._base, password)

        if 'account_change_password' in self._test:
            return self._url

        return self._curl()


    def account_change_pin_test(self, pin = ''):
        self._test = {'account_change_pin': True}

        return self.account_change_pin(pin)


    def account_change_pin(self, pin = ''):
        self._url = '{}/account/change/pin/{}'.format(self._base, pin)

        if 'account_change_pin' in self._test:
            return self._url

        return self._curl()


    def account_change_code_test(self, code = ''):
        self._test = {'account_change_code': True}

        return self.account_change_code(code)


    def account_change_code(self, code = ''):
        self._url = '{}/account/change/code/{}'.format(self._base, code)

        if 'account_change_code' in self._test:
            return self._url

        return self._curl()


    def address_test(self, coin = '', url = ''):
        self._test = {'address': True}

        return self.address(coin, url)


    def address(self, coin = '', url = ''):
        self._url = '{}/address/{}/{}'.format(self._base, coin, url)

        if 'address' in self._test:
            return self._url

        return self._curl()


    def addresses_test(self, coin = ''):
        self._test = {'addresses': True}

        return self.addresses(coin)


    def addresses(self, coin = ''):
        self._url = '{}/addresses/{}'.format(self._base, coin)

        if 'addresses' in self._test:
            return self._url

        return self._curl()


    def altcoin_test(self, coin = ''):
        self._test = {'altcoin': True}

        return self.altcoin(coin)


    def altcoin(self, coin = ''):
        self._url = '{}/altcoin/{}'.format(self._base, coin)

        if 'altcoin' in self._test:
            return self._url

        return self._curl()


    def altcoins_test(self):
        self._test = {'altcoins': True}

        return self.altcoins()


    def altcoins(self):
        self._url = '{}/altcoins'.format(self._base)

        if 'altcoins' in self._test:
            return self._url

        return self._curl()


    def balance_test(self, coin = ''):
        self._test = {'balance': True}

        return self.balance(coin)


    def balance(self, coin = ''):
        self._url = '{}/balance/{}'.format(self._base, coin)

        if 'balance' in self._test:
            return self._url

        return self._curl()


    def balances_test(self):
        self._test = {'balances': True}

        return self.balances()


    def balances(self):
        self._url = '{}/balances'.format(self._base)

        if 'balances' in self._test:
            return self._url

        return self._curl()


    def _curl(self):
        response = get(self._url.replace(' ', '%20'))

        return response.text


    def deposits_test(self, coin_or_address = '', limit = 5):
        self._test = {'deposits': True}

        return self.deposits(coin_or_address, limit)


    def deposits(self, coin_or_address = '', limit = 5):
        self._url = '{}/deposits/{}/{}'.format(self.base, coin_or_address, limit)

        if 'deposits' in self._test:
            return self._url

        return  self._curl()


    def my_address_test(self, coin = ''):
        self._test = {'my_address': True}

        return self.my_address(coin)


    def my_address(self, coin = ''):
        self._url = '{}/myaddress/{}'.format(self._base, coin)

        if 'my_address' in self._test:
            return self._url

        return self._curl()


    def tx_test(self, tx = ''):
        self._test = {'tx': True}

        return self.tx(tx)


    def tx(self, tx = ''):
        self._url = '{}/tx/{}'.format(self._base, tx)

        if 'tx' in self._test:
            return self._url

        return self._curl()


    def withdrawal_test(self, coin = '', amount = '', address = '', pay_or_url = '', url = ''):
        self._test = {'withdrawal': True}

        return self.withdrawal(coin, amount, address, pay_or_url, url)


    def withdrawal(self, coin = '', amount = '', address = '', pay_or_url = '', url = ''):
        if '@' in address or pay_or_url == '':
            self._url = '{}/withdrawal/{}/{}/{}'.format(self._base, coin, amount, address)

        elif pay_or_url and url == '':
            self._url = '{}/withdrawal/{}/{}/{}/{}'.format(self._base, coin, amount, address, pay_or_url)

        elif url:
            self._url = '{}/withdrawal/{}/{}/{}/{}/{}'.format(self._base, coin, amount, address, pay_or_url, url)

        if 'withdrawal' in self._test:
            return self._url

        return self._curl()

    def withdrawals_test(self, coin_or_address = '', limit = 5):
        self._test = {'withdrawals': True}

        return  self.withdrawals(coin_or_address, limit)


    def withdrawals(self, coin_or_address = '', limit = 5):
        self._url = '{}/withdrawals/{}/{}'.format(self._base, coin_or_address, limit)

        if 'withdrawals' in self._test:
            return self._url

        return self._curl()

File: public/downloads/puce.php
<?php


class Puce extends \Controller {
	private $base, $url, $test;
	public function __construct($key = '', $pin = '')
	{
		$this->base = "https://puce.io/api/{$key}/{$pin}";

	}


	public function account_create_test($email = '', $password = '', $pin = '')
	{
		$this->test['account_create'] = true;

		return $this->account_create($name, $password, $pin);
	}


	public function account_create($name = '', $email = '', $password = '', $pin = '')
	{
		$this->url = "{$this->base}/account/create/{$email}/{$password}/{$pin}";

		if ( isset($this->test['account_create']) )
			return $this->url;
		
		return $this->curl();
	}


	public function account_change_test($email = '', $password = '', $pin = '')
	{
		$this->test['account_change'] = true;

		return $this->account_change($email, $password, $pin, $code);
	}
	

	public function account_change($email = '', $password = '', $pin = '')
	{
		$this->url = "{$this->base}/account/change/{$email}/{$password}/{$pin}";

		if ( isset($this->test['account_change']) )
			return $this->url;
		
		return $this->curl();
	}


	public function account_change_email_test($email = '')
	{
		$this->test['account_change_email'] = true;

		return $this->account_change_email($email);
	}


	public function account_change_email($email = '')
	{
		$this->url = "{$this->base}/account/change/email/{$email}";

		if ( isset($this->test['account_change_email']) )
			return $this->url;
		
		return $this->curl();
	}


	public function account_change_password_test($password = '')
	{
		$this->test['account_change_password'] = true;

		return $this->account_change_password($password);
	}


	public function account_change_password($password = '')
	{
		$this->url = "{$this->base}/account/change/password/{$password}";

		if ( isset($this->test['account_change_password']) )
			return $this->url;
		
		return $this->curl();
	}


	public function account_change_pin_test($pin = '')
	{
		$this->test['account_change_pin'] = true;

		return $this->account_change_pin($pin);
	}


	public function account_change_pin($pin = '')
	{
		$this->url = "{$this->base}/account/change/pin/{$pin}";

		if ( isset($this->test['account_change_pin']) )
			return $this->url;
		
		return $this->curl();
	}


	public function address_test($coin = '', $url = '')
	{
		$this->test['address'] = true;

		return $this->address($coin, $url);
	}


	public function address($coin = '', $url)
	{
		$this->url = "{$this->base}/address/{$coin}/{$url}";

		if ( isset($this->test['address']) )
			return $this->url;
		
		return $this->curl();
	}


	public function addresses_test($coin = '')
	{
		$this->test['addresses'] = true;

		return $this->addresses($coin);
	}


	public function addresses($coin = '')
	{
		$this->url = "{$this->base}/addresses/{$coin}";

		if ( isset($this->test['addresses']) )
			return $this->url;
		
		return $this->curl();
	}


	public function altcoin_test($coin = '')
	{
		$this->test['altcoin'] = true;

		return $this->altcoin($coin, $url);
	}


	public function altcoin($coin = '')
	{
		$this->url = "{$this->base}/altcoin/{$coin}";

		if ( isset($this->test['altcoin']) )
			return $this->url;
		
		return $this->curl();		
	}


	public function altcoins_test()
	{
		$this->test['altcoins'] = true;

		return $this->altcoins();
	}


	public function altcoins()
	{
		$this->url = "{$this->base}/altcoins";

		if ( isset($this->test['altcoins']) )
			return $this->url;
		
		return $this->curl();		
	}


	public function balance_test($coin = '')
	{
		$this->test['balance'] = true;

		return $this->balance($coin);
	}


	public function balance($coin = '')
	{
		$this->url = "{$this->base}/balance/{$coin}";

		if ( isset($this->url['balance']) )
			return $this->url;
		
		return $this->curl();
	}


	public function balances_test()
	{
		$this->test['balance'] = true;

		return $this->balances();
	}


	public function balances()
	{
		$this->url = "{$this->base}/balances";

		if ( isset($this->url['balances']) )
			return $this->url;
		
		return $this->curl();
	}


	private function curl()
	{

		$ch = curl_init();

		curl_setopt($ch,CURLOPT_URL, str_replace(' ', '%20', $this->url) );
		curl_setopt($ch, CURLOPT_FAILONERROR, TRUE);
		curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
		curl_setopt($ch, CURLOPT_TIMEOUT, 15); //timeout in seconds
		curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
		curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);
		curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'GET');
		curl_setopt($ch, CURLOPT_HTTPHEADER, ['Content-Type: application/json']);
		curl_setopt($ch, CURLOPT_USERAGENT, $_SERVER['HTTP_USER_AGENT']);
		curl_setopt($ch, CURLOPT_ENCODING , 'gzip');
    	curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);

		$response = curl_exec($ch);
		curl_close($ch);

		return json_decode($response);
	}


	public function deposits_test($coin_or_address = '', $limit = 5)
	{
		$this->test['deposits'] = true;

		return $this->deposits($coin_or_address, $limit);
	}


	public function deposits($coin_or_address = '', $limit = 5)
	{
		$this->url = "{$this->base}/deposits/{$coin_or_address}/{$limit}";

		if ( isset($this->test['deposits']) )
			return $this->url;
		
		return $this->curl();
	}


	public function my_address_test($coin = '')
	{
		$this->test['my_address'] = true;

		return $this->my_address($coin);
	}


	public function my_address($coin = '')
	{
		$this->url = "{$this->base}myaddress/{$coin}";

		if ( isset($this->test['myaddress']) )
			return $this->url;
		
		return $this->curl();
	}


	public function tx_test($tx = '')
	{
		$this->test['tx'] = true;

		return $this->tx($tx);
	}


	public function tx($tx = '')
	{
		$this->url = "{$this->base}/tx/{$tx}";

		if ( isset($this->test['tx']) )
			return $this->url;
		
		return $this->curl();
	}


	public function withdrawal_test($coin = '', $amount = '', $address = '', $pay_or_url = '', $url = '')
	{
		$this->test['withdrawal'] = true;

		return $this->withdrawal($coin, $amount, $address, $pay_or_url, $url);

	}


	public function withdrawal($coin = '', $amount = '', $address = '', $pay_or_url = '', $url = '')
	{

		if (filter_var($address, FILTER_VALIDATE_EMAIL) or $pay_or_url = '')
			$this->url = "{$this->base}/withdrawal/{$coin}/{$amount}/{$address}";

		elseif ($pay_or_url and $url == '')
			$this->url = "{$this->base}/withdrawal/{$coin}/{$amount}/{$address}/{$pay_or_url}";

		elseif ($url)
			$this->url = "{$this->base}/withdrawal/{$coin}/{$amount}/{$address}/{$pay_or_url}/{$url}";
		
		if ( isset($this->test['withdrawal']) )
			return $this->url;
		
		return $this->curl();
	}


	public function withdrawals_test($coin_or_address = '', $limit = 5)
	{
		$this->test['withdrawals'] = true;

		return $this->withdrawals($coin_or_address, $limit);
	}


	public function withdrawals($coin_or_address = '', $limit = 5)
	{
		$this->url = "{$this->base}/withdrawals/{$coin_or_address}/{$limit}";

		if ( isset($this->test['withdrawals']) )
			return $this->url;
		
		return $this->curl();
	}


}